sql/src/test/java/io/crate/execution/engine/join/HashInnerJoinBatchIteratorMemoryTest.java

```diff
@@ -89,26 +91,51 @@ public void testReleaseAccountingRows() throws Exception {
     public void testCalculationOfBlockSize() {
         when(circuitBreaker.getLimit()).thenReturn(110L);
         when(circuitBreaker.getUsed()).thenReturn(10L);
-        assertThat(calculateBlockSize(circuitBreaker, 5, 100), is(20));
-        assertThat(calculateBlockSize(circuitBreaker, 5, 10), is(10));
+        assertThat(calculateBlockSize(circuitBreaker, 5, 100, -1, false), is(20));
+        assertThat(calculateBlockSize(circuitBreaker, 5, 10, -1, false), is(10));
+    }
+
+    @Test
+    public void testBlockSizeIsAdjustedForUnorderedLimitJoins() {
+        when(circuitBreaker.getLimit()).thenReturn(HashInnerJoinBatchIterator.DEFAULT_BLOCK_SIZE + 1L);
+        when(circuitBreaker.getUsed()).thenReturn(0L);
+        long numberOfRowsForLeft = HashInnerJoinBatchIterator.DEFAULT_BLOCK_SIZE;
```

Comment:
shouldn't this be greater than the default block size to validate the behaviour?